### ISSUE TITLE
[feature request] PreComputation of a layer's output 

### ISSUE THREAD

**Issue by 50417** [pytorch/pytorch#5515]
Just like Tensor.detach_() function prevents from backward propagation and weight update of the layer. A function that would precompute the layer's output when it has been detached making the neural net smaller for testing would be beneficial for research work even if it will blow up memory when tensor are huge.

**Comment by SsnL** [comment]
Could you clarify what you mean by precomputing a layer's output?

**Comment by 50417** [comment]
Lets say I am training a CNN model 

While training the model, if early layer of the net converges, so i detach it from backprop. Since any input through that detached layer is essentially produce same output no matter the number of future epochs, its redundant.  I am requesting for a feature such that such layer's output is precomputed and fed directly to the consecutive layers rather than computing each time.

**Comment by apaszke** [comment]
While certainly possible such output caching wouldn't provide a lot of speedup, and would consume a ton of disk space. It's likely that reading the data from disk would be slower than recomputing it. Thanks for the proposal, but I don't think we'll be adding this feature.

**Comment by 50417** [comment]
Practical application with huge data set might be limited. Since, a lot of researcher use pytorch who might have vast amount of resources at their disposal, such feature might be handy assuming everything is stored in GPU memory.
